amr-wind/equation_systems/icns/source_terms/DragForcing.cpp

```diff
@@ -0,0 +1,120 @@
+#include "amr-wind/equation_systems/icns/source_terms/DragForcing.H"
+#include "amr-wind/utilities/IOManager.H"
+
+#include "AMReX_Gpu.H"
+#include "AMReX_Random.H"
+#include "amr-wind/wind_energy/ABL.H"
+
+namespace amr_wind::pde::icns {
+
+DragForcing::DragForcing(const CFDSim& sim)
+    : m_sim(sim)
+    , m_mesh(sim.mesh())
+    , m_velocity(sim.repo().get_field("velocity"))
+    , m_terrainBlank(sim.repo().get_field("terrainBlank"))
+    , m_terrainDrag(sim.repo().get_field("terrainDrag"))
+
+{
+    const auto& abl = m_sim.physics_manager().get<amr_wind::ABL>();
```

Comment:
if this requires `ABL` should we add a check for it's existence? And maybe rename it to something that signifies that this is an ABL drag forcing?